### ISSUE TITLE
[TTS] Why VITS consumes  so large amount of CUDA memory?

### ISSUE THREAD

**Issue by zzxiang** [espnet/espnet#3719]
I tried to fine-tune VITS with [egs2/jsut/tts1/conf/tuning/
train_full_band_vits.yaml ](https://github.com/espnet/espnet/blob/master/egs2/jsut/tts1/conf/tuning/train_full_band_vits.yaml) on a small data of 200 wav files, but failed to start stage 6 because of out of CUDA memory.

According to the comment at the beginning of the yaml file, the training of VITS on JSUT and the fine-tuning of VITS on JVS both succeeded on 4 GPUs (V100) with 32GB GPU memory. On the other hand, my machine is equipped only with a Tesla T4 GPU with 15GB GPU memory, apparently having a much smaller GPU memory amount.

I wonder why VITS costs so much GPU memory.

Is it possible to do the fine-tuning of VITS on a GPU with relatively small amount of GPU memory, for example, by modify some parameters in the yaml file? If it is possible, does it affect the quality of the synthesis result, or the accuracy of training? Or does it only effect the training time?

**Comment by kan-bayashi** [comment]
Actually, it does not fully utilize 32GB memory (around ~20GB).
You can reduce the batch_bins (say, 5000000).
https://github.com/espnet/espnet/blob/51f267310c6159e688d3f1cf66353f65a14effef/egs2/tsukuyomi/tts1/conf/tuning/finetune_full_band_vits.yaml#L163

> If it is possible, does it affect the quality of the synthesis result, or the accuracy of training? Or does it only effect the training time?

If the actual batch size is not too small, maybe it works.
(You can see the actual batch size in `train.log`.)
```
[huracan:0/4] 2021-09-12 09:30:11,679 (abs_task:1493) INFO: [train] dataset:
ESPnetDataset(
  text: {"path": "dump/44.1k/raw/tr_no_dev/text", "type": "text"}
  speech: {"path": "dump/44.1k/raw/tr_no_dev/wav.scp", "type": "sound"}
  preprocess: <espnet2.train.preprocessor.CommonPreprocessor object at 0x2b0bcec809e8>)
[huracan:0/4] 2021-09-12 09:30:11,679 (abs_task:1494) INFO: [train] Batch sampler: NumElementsBatchSampler(N-batch=207, batch_bins=7000000, sort_in_b
atch=descending, sort_batch=descending)
[huracan:0/4] 2021-09-12 09:30:11,679 (abs_task:1496) INFO: [train] mini-batch sizes summary: N-batch=207, mean=34.8, min=13, max=79
```

**Comment by zzxiang** [comment]
Thank you! I reduced the batch_bins to 3000000 and it works!

**Comment by fijipants** [comment]
Had same issue when training on LJSpeech on RTX 3090 (24GB), `3000000` broke for me but `2500000` worked.
